pkg/sources/git/git.go

```diff
@@ -1241,32 +1241,44 @@ func (s *Git) handleBinary(ctx context.Context, gitDir string, reporter sources.
 	}
 
 	cmd := exec.Command("git", "-C", gitDir, "cat-file", "blob", commitHash.String()+":"+path)
-	stdout, err := s.executeCatFileCmd(cmd)
+	stdout, catCmd, err := s.executeCatFileCmd(cmd)
 	if err != nil {
 		return err
 	}
+	// Wait must be called after closing the pipe (defer is a stack, so first defer is executed last)
+	defer func() {
+		_ = catCmd.Wait()
+	}()
+	defer stdout.Close()
+
+	err = handlers.HandleFile(ctx, stdout, chunkSkel, reporter, handlers.WithSkipArchives(s.skipArchives))
 
-	if err = handlers.HandleFile(ctx, stdout, chunkSkel, reporter, handlers.WithSkipArchives(s.skipArchives)); err != nil {
+	// Always call Wait() to ensure the process is properly cleaned up
+	waitErr := cmd.Wait()
```

Comment:
This seems like a bug. `cmd` and `catCmd` are the same thing. It seems that `cmd.Wait()` won't exit until `stdout.Close()` is called, but that won't run into `cmd.Wait()` is done.

See: #3374

Reply:
Oops! Thank you for digging into this.

Reply:
Doesn't really seem like there is benefit in executeCatFileCmd being a separate function IMO, it's only used once anyway.